src/sentry/integrations/slack/message_builder/issues.py

```diff
@@ -58,6 +58,18 @@
 from sentry.utils import json
 
 STATUSES = {"resolved": "resolved", "ignored": "ignored", "unresolved": "re-opened"}
+PROVIDER_TO_COMMIT_LINK_URL_FORMAT = {
+    "github": "{base_url}/commit/{commit_id}",
+    "integrations:github": "{base_url}/commit/{commit_id}",
+    "integrations:github_enterprise": "{base_url}/commit/{commit_id}",
+    "visualstudio": "{base_url}/commit/{commit_id}",
+    "integrations:vsts": "{base_url}/commit/{commit_id}",
+    "gitlab": "{base_url}/commit/{commit_id}",
+    "integrations:gitlab": "{base_url}/commit/{commit_id}",
+    "bitbucket": "{base_url}/commits/{commit_id}",
+    "integrations:bitbucket": "{base_url}/commits/{commit_id}",
+}
```

Comment:
ill admit there might be a better way of tracking this 😬

Reply:
What if you removed all the dupes / integration starting ones like:

```
SUPPORTED_PROVIDER_TO_COMMIT_URL = {
    "github": "{base_url}/commit/{commit_id}",
    "vsts": "{base_url}/commit/{commit_id}",
    "gitlab": "{base_url}/commit/{commit_id}",
    "bitbucket": "{base_url}/commits/{commit_id}",
}
```

and then where they're being checked instead of:

```
        if repo_base and provider in SUPPORTED_PROVIDER_TO_COMMIT_URL:
            commit_link = f"<{SUPPORTED_PROVIDER_TO_COMMIT_URL[provider].format(base_url=repo_base, commit_id=commit_id)}|{commit_id[:6]}>"
            suspect_commit_text += f"{commit_link} by {author_display}"
```

you did:

```
if repo_base:
    for supported_provider, url in SUPPORTED_PROVIDER_TO_COMMIT_URL.items():
        if supported_provider in provider:
            commit_link = f"<{url.format(base_url=repo_base, commit_id=commit_id)}|{commit_id[:6]}>"
```

or since bitbucket is the only different one maybe like:

```
if repo_base:
   if "bitbucket" in provider:
        commit_link = f"{base_url}/commits/{commit_id}"
    else:
        ...
```